pallets/capacity/src/lib.rs

```diff
@@ -637,18 +644,20 @@ impl<T: Config> Pallet<T> {
 			Error::<T>::StakingAmountBelowMinimum
 		);
 
-		Ok((staking_account, stakable_amount))
+		Ok((staking_details, stakable_amount))
 	}
 
-	// TODO: this should return StakingAccount, BoostHistory and Balance
 	fn ensure_can_boost(
 		staker: &T::AccountId,
 		target: &MessageSourceId,
 		amount: &BalanceOf<T>,
 	) -> Result<(StakingDetails<T>, BalanceOf<T>), DispatchError> {
-		let (staking_details, stakable_amount) = Self::ensure_can_stake(staker, *target, *amount)?;
-		// TODO: boost history
-		// let boost_history = Self::get_boost_history_for(staker).unwrap_or_defaul();
+		// FIXME: if one is boosting additional amounts, this will fail.
+		let (mut staking_details, stakable_amount) = 
+			Self::ensure_can_stake(staker, *target, *amount, ProviderBoost)?;
+		staking_details.staking_type = ProviderBoost;
```

Comment:
We do need to set it specifically if it's zero because this will come back as the default of MaximumCapacity.  The bug is when you are calling provider boost and the account is found, it will not have active == 0. Then it will break. I intend to fix this code properly in a different PR.  I just wanted to get tests passing and set the 50% value.